index_service/utils.py

```diff
@@ -1,9 +1,129 @@
 
+import functools
+import logging
 import random
 
+from django.conf import settings
+from django.http import JsonResponse
+from django.utils import translation
+
+import requests
+
+import phonenumbers
+
 
 def short_id(length):
     CHARSET = 'CDEHKMPRSTUWXY2458'
     get_character = CHARSET.__getitem__
     random_numbers = (random.randrange(len(CHARSET)) for i in range(length))
     return ''.join(map(get_character, random_numbers))
+
+
+def parse_phone_number(phone_number, fallback_cc):
+    try:
+        return phonenumbers.parse(phone_number, region=fallback_cc)
+    except phonenumbers.NumberParseException as exc:
+        raise ValueError('Unable to parse phone number %r: %s' % (phone_number, exc)) from exc
+
+
+def normalize_phone_number(phone_number, fallback_cc):
+    """
+    Return (str) *phone_number* (str) normalized to ITU-T E.164.
+
+    Apply fallback_CC (str/None) country code, if necessary.
+    """
+    phone_number = parse_phone_number(phone_number, fallback_cc)
+    return phonenumbers.format_number(phone_number, phonenumbers.PhoneNumberFormat.E164)
+
+
+def get_current_cc(language=None):
+    language = language or translation.get_language()
+    return language.split('-')[-1].upper()
+
+
+def normalize_phone_number_localised(phone_number):
+    return normalize_phone_number(phone_number, get_current_cc())
+
+
+logger = logging.getLogger('index_service.utils.authorization')
+
+
+def check_authorization(request):
+    if not settings.REQUIRE_AUTHORIZATION:
+        reason = 'No authorization required, none checked.'
+        logger.info('Request is authorized: %s', reason)
+        return True, reason
+    auth_header = request.META.get('HTTP_AUTHORIZATION')
+    if not auth_header:
+        reason = 'No authorization supplied.'
+        logger.warning('Request is unauthorized: %s', reason)
+        return False, reason
+    acked, reason = AccountingAuthorization().check(auth_header)
+    if not acked:
+        logger.warning('Request is unauthorized: %s', reason)
+        return False, reason
+    logger.info('Request is authorized: %s', reason)
+    return True, reason
+
+
+class AccountingAuthorization:
```

Comment:
This class is framework-agnostic and can be pulled later into a separate package for re-use, if needed.